### ISSUE TITLE
execution order defined scripts

### ISSUE THREAD

**Issue by sdeedee** [newrelic/nri-flex#55]
Hi,

I have a question about the execution order of scripts nri-flex is using.

Today we are using the nagios OHI to execute some scripts/checks but I noticed the nagios OHI is executing defined scripts/checks serial. For example if we have a script defined for 10 assets. The script will run asset per asset. If a script is timing out on for example asset 5. Asset 6 - 10 the script will not be executed waiting on the timeout on asset 5. This means in some cases we don't have samples/metrics in some time windows ... what's not good.

I was wondering how the nri-flex is executing scripts/commands that should run on multiple hosts. Also serial? Or will the nri-flex starts the script/command in parallel over all defined hosts?

Thanks.

**Comment by Kav91** [comment]
Hi, 
Say you have 3 Flex config files stored in the directory, all 3 will execute asynchronously and obey the default timeouts unless otherwise increased.

Also if you use a [lookup file](https://github.com/newrelic/nri-flex/wiki/Functions#lookup_file), that will create multiple configs which will execute async.

However say you have multiple steps within your Flex config,  they are done synchronously/sequentially in the order they have been defined.

So you have the option to structure either way depending on your needs.

**Comment by Kav91** [comment]
Closing question answered & over 1 week inactive.
Feel free to create another issue for any further questions.
Thanks
